gnd/src/main/java/com/google/android/gnd/rx/annotations/Cold.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.rx.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+/**
+ * Marks an observable as "cold". Cold observables create a new producer each time an observer
+ * subscribes.
+ *
+ * <p>Cold observables start emitting items when observer subscribe, so observers are guaranteed to
+ * see the whole sequence from the beginning.
+ *
+ * <p>Since a new producer is created on each subscription, the sequence of items is regenerated for
+ * each observer. As such, cold observables are unsuitable for multicasting.
+ *
+ * <p>LiveData and Rx observables backed by cold observables are also cold. New cold Rx observables
+ * may also be created by emitting results inside the class passed to the observable's <code>
+ * create()</code> method.
+ *
+ * <p>Cold Rx observables may turned into hot connectable observers using their respective <code>
```

Comment:
Actually removed this altogether; would prefer to find other solutions to share(), since it leads to "warm" observables that are hard to reason about in more complex systems, imo.